main/src/ca/uwaterloo/flix/language/phase/MonoTypes.scala

```diff
@@ -382,34 +290,31 @@ object MonoTypes {
   }
 
   /**
-    * Returns a type with specialized enums and no aliases.
+    * Returns a type where 1) aliases are removed and 2) enums are non-parametric.
     */
-  private def visitType(tpe: Type)(implicit ctx: SharedContext, root: LoweredAst.Root, flix: Flix): Type = {
-    def visitInner(tpe0: Type): Type = tpe0.baseType match {
-      case Type.Cst(TypeConstructor.Enum(sym, _), loc) => // enum
-        // args cannot be visited yet, because specialization works on
-        // non-specialized enums.
-        val args = tpe0.typeArguments
-        val freshSym = specializeEnum(sym, args, loc)
-        // visit args
-        args.foreach(visitInner)
-        Type.mkEnum(freshSym, Nil, loc)
-      case _ => tpe0 match { // non-enum
-        case Type.Cst(tc, loc) => Type.Cst(tc, loc)
-        case Type.Apply(tpe1, tpe2, loc) => Type.Apply(visitInner(tpe1), visitInner(tpe2), loc)
-        case Type.Var(sym, loc) => throw InternalCompilerException(s"Unexpected type var: '$sym'", loc)
-        case Type.Alias(cst, _, _, loc) => throw InternalCompilerException(s"Unexpected type alias: '${cst.sym}'", loc)
-        case Type.AssocType(cst, _, _, loc) => throw InternalCompilerException(s"Unexpected associated type: '${cst.sym}'", loc)
-      }
-    }
-    // It is important that eraseAliases happens BEFORE enum specialization
-    visitInner(eraseAliases(tpe))
+  private def visitType(tpe: Type): Type = tpe match {
+    case Type.Cst(TypeConstructor.Enum(sym, _), loc) =>
+      // Remove type arguments from enums.
+      Type.Cst(TypeConstructor.Enum(sym, Kind.Star), loc)
+    case Type.Cst(tc, loc) =>
+      Type.Cst(tc, loc)
+    case Type.Apply(tpe1, tpe2, loc) =>
+      Type.Apply(visitType(tpe1), visitType(tpe2), loc)
```

Comment:
@JonathanStarup I will probably merge this now, but you might want to check if type arguments are actually erased, because at least here you rebuild the Type.Apply which is how we represent type arguments.